### PR TITLE
Refactor tests under `limiter` package.

### PR DESCRIPTION
**Pull request by russjones**
Refactored all tests under `lib/limiter` to use testify instead of gocheck.

### CODE REVIEW

File: lib/limiter/limiter_test.go [hunk 1]
```diff
@@ -21,42 +21,34 @@ import (
...
-
-func (s *LimiterSuite) TestConnectionsLimiter(c *C) {
+func TestConnectionsLimiter(t *testing.T) {
 	limiter, err := NewLimiter(
 		Config{
 			MaxConnections: 0,
 		},
 	)
-	c.Assert(err, IsNil)
+	require.Nil(t, err)
```
Comment (zmb3) [inline lib/limiter/limiter_test.go:45]: Use NoError here instead?

File: lib/limiter/limiter_test.go [hunk 2]
```diff
@@ -119,28 +111,28 @@ func (s *LimiterSuite) TestRateLimiter(c *C) {
 				},
 			},
 		})
-	c.Assert(err, IsNil)
+	require.Nil(t, err)
```
Comment (zmb3) [inline lib/limiter/limiter_test.go:114]: NoError

File: lib/limiter/limiter_test.go [hunk 3]
```diff
@@ -119,28 +111,28 @@ func (s *LimiterSuite) TestRateLimiter(c *C) {
...
-		c.Assert(limiter.RegisterRequest("token1"), IsNil)
+		require.NoError(t, limiter.RegisterRequest("token1"))
 	}
 	for i := 0; i < 20; i++ {
-		c.Assert(limiter.RegisterRequest("token2"), IsNil)
+		require.NoError(t, limiter.RegisterRequest("token2"))
 	}
 
-	c.Assert(limiter.RegisterRequest("token1"), NotNil)
+	require.NotNil(t, limiter.RegisterRequest("token1"))
```
Comment (zmb3) [inline lib/limiter/limiter_test.go:123]: require.Error

File: lib/limiter/limiter_test.go [hunk 4]
```diff
@@ -71,33 +63,33 @@ func (s *LimiterSuite) TestConnectionsLimiter(c *C) {
 			MaxConnections: 5,
 		},
 	)
-	c.Assert(err, IsNil)
+	require.Nil(t, err)
```
Comment (r0mant) [inline lib/limiter/limiter_test.go:66]:
```suggestion
	require.NoError(t, err)
```

File: lib/limiter/limiter_test.go [hunk 5]
```diff
@@ -119,28 +111,28 @@ func (s *LimiterSuite) TestRateLimiter(c *C) {
...
-	c.Assert(limiter.RegisterRequest("token1"), NotNil)
+	require.Error(t, limiter.RegisterRequest("token1"))
 
 	clock.Sleep(10 * time.Millisecond)
 	for i := 0; i < 10; i++ {
-		c.Assert(limiter.RegisterRequest("token1"), IsNil)
+		require.NoError(t, limiter.RegisterRequest("token1"))
 	}
-	c.Assert(limiter.RegisterRequest("token1"), NotNil)
+	require.NotNil(t, limiter.RegisterRequest("token1"))
```
Comment (r0mant) [inline lib/limiter/limiter_test.go:129]:
```suggestion
	require.Error(t, limiter.RegisterRequest("token1"))
```

File: lib/limiter/limiter_test.go [hunk 6]
```diff
@@ -119,28 +111,28 @@ func (s *LimiterSuite) TestRateLimiter(c *C) {
...
 
 	clock.Sleep(10 * time.Millisecond)
 	for i := 0; i < 10; i++ {
-		c.Assert(limiter.RegisterRequest("token1"), IsNil)
+		require.NoError(t, limiter.RegisterRequest("token1"))
 	}
-	c.Assert(limiter.RegisterRequest("token1"), NotNil)
+	require.NotNil(t, limiter.RegisterRequest("token1"))
```
Comment (r0mant) [inline lib/limiter/limiter_test.go:135]:
```suggestion
	require.Error(t, limiter.RegisterRequest("token1"))
```

File: lib/limiter/limiter_test.go [hunk 7]
```diff
@@ -151,22 +143,22 @@ func (s *LimiterSuite) TestRateLimiter(c *C) {
 			break
 		}
 	}
-	c.Assert(err, NotNil)
+	require.NotNil(t, err)
```
Comment (r0mant) [inline lib/limiter/limiter_test.go:146]:
```suggestion
	require.Error(t, err)
```

File: lib/limiter/limiter_test.go [hunk 8]
```diff
@@ -151,22 +143,22 @@ func (s *LimiterSuite) TestRateLimiter(c *C) {
...
+	require.NoError(t, limiter.RegisterRequest("token1"))
 	err = nil
 	for i := 0; i < 15; i++ {
 		err = limiter.RegisterRequest("token1")
 		if err != nil {
 			break
 		}
 	}
-	c.Assert(err, NotNil)
+	require.NotNil(t, err)
```
Comment (r0mant) [inline lib/limiter/limiter_test.go:158]:
```suggestion
	require.Error(t, err)
```

File: lib/limiter/limiter_test.go [hunk 9]
```diff
@@ -183,23 +175,23 @@ func (s *LimiterSuite) TestCustomRate(c *C) {
 				},
 			},
 		})
-	c.Assert(err, IsNil)
+	require.Nil(t, err)
```
Comment (r0mant) [inline lib/limiter/limiter_test.go:178]:
```suggestion
	require.NoError(t, err)
```

File: lib/limiter/limiter_test.go [hunk 10]
```diff
@@ -183,23 +175,23 @@ func (s *LimiterSuite) TestCustomRate(c *C) {
...
 
 	customRate := ratelimit.NewRateSet()
 	err = customRate.Add(time.Minute, 1, 5)
-	c.Assert(err, IsNil)
+	require.Nil(t, err)
```
Comment (r0mant) [inline lib/limiter/limiter_test.go:182]:
```suggestion
	require.NoError(t, err)
```

File: lib/limiter/limiter_test.go [hunk 11]
```diff
@@ -183,23 +175,23 @@ func (s *LimiterSuite) TestCustomRate(c *C) {
...
 
 	// Max out custom rate.
 	for i := 0; i < 5; i++ {
-		c.Assert(limiter.RegisterRequestWithCustomRate("token1", customRate), IsNil)
+		require.NoError(t, limiter.RegisterRequestWithCustomRate("token1", customRate))
 	}
 
 	// Test rate limit exceeded with custom rate.
-	c.Assert(limiter.RegisterRequestWithCustomRate("token1", customRate), NotNil)
+	require.NotNil(t, limiter.RegisterRequestWithCustomRate("token1", customRate))
```
Comment (r0mant) [inline lib/limiter/limiter_test.go:190]:
```suggestion
	require.Error(t, limiter.RegisterRequestWithCustomRate("token1", customRate))
```

File: lib/limiter/limiter_test.go [hunk 12]
```diff
@@ -71,33 +63,33 @@ func (s *LimiterSuite) TestConnectionsLimiter(c *C) {
...
+		require.NoError(t, limiter.AcquireConnection("token1"))
 	}
 
 	for i := 0; i < 5; i++ {
-		c.Assert(limiter.AcquireConnection("token2"), IsNil)
+		require.NoError(t, limiter.AcquireConnection("token2"))
 	}
 	for i := 0; i < 5; i++ {
-		c.Assert(limiter.AcquireConnection("token2"), NotNil)
+		require.NotNil(t, limiter.AcquireConnection("token2"))
```
Comment (r0mant) [inline lib/limiter/limiter_test.go:76]:
```suggestion
		require.Error(t, limiter.AcquireConnection("token2"))
```

**Review by r0mant — APPROVED** [review]

**Review by rosstimothy — APPROVED** [review]

**Review by zmb3 — APPROVED** [review]

**Pull request merged**